std/array.d

```diff
@@ -2146,13 +2151,26 @@ if (isDynamicArray!(E[]) && isForwardRange!R1 && isForwardRange!R2
     assert("Hello Wörld".replace("l", "h") == "Hehho Wörhd");
 }
 
+// https://issues.dlang.org/show_bug.cgi?id=18215
+@safe unittest
+{
+    auto arr = ["aaa.dd", "b"];
+    arr = arr.replace("aaa.dd", ".");
+    assert(arr == [".", "b"]);
+
+    arr = ["_", "_", "aaa.dd", "b", "c", "aaa.dd", "e"];
+    arr = arr.replace("aaa.dd", ".");
+    assert(arr == ["_", "_", ".", "b", "c", ".", "e"]);
+}
```

Comment:
Is there already a unittest that covers the case `"abababc".replace("abcdef","g")`? I.e., the second branch of the static if.

Reply:
Yeah that's the original case and covered by the existing tests. Expand the diff above.

![image](https://user-images.githubusercontent.com/4370550/34733674-7cc60704-f569-11e7-838a-a9fb1923a759.png)

BTW you might want to install the CodeCov browser extension: https://github.com/codecov/browser-extension

Reply:
I still added more tests ;-)

Reply:
Yeah I was confused by your change to use `min(from.length, $)`, which you just reverted. I suppose that was not necessary? :-P